# bakta container

Main tool: [bakta](https://github.com/oschwengers/bakta)


<details>
<summary>Additional tools installed via micromamba:</summary>

```
  Name                     Version       Build                 Channel    
────────────────────────────────────────────────────────────────────────────
  _libgcc_mutex            0.1           conda_forge           conda-forge
  _openmp_mutex            4.5           2_gnu                 conda-forge
  about-time               4.2.1         pyhd8ed1ab_1          conda-forge
  alive-progress           3.0.1         pyhd8ed1ab_0          conda-forge
  aragorn                  1.2.41        h7b50bb2_5            bioconda   
  archspec                 0.2.5         pyhd8ed1ab_0          conda-forge
  backports.zstd           1.3.0         py313h18e8e13_0       conda-forge
  bakta                    1.12.0        pyhdfd78af_0          bioconda   
  biopython                1.86          py313h07c4f96_1       conda-forge
  blast                    2.17.0        h66d330f_0            bioconda   
  brotli                   1.2.0         hed03a55_1            conda-forge
  brotli-bin               1.2.0         hb03c661_1            conda-forge
  brotli-python            1.2.0         py313hf159716_1       conda-forge
  bzip2                    1.0.8         hda65f42_8            conda-forge
  c-ares                   1.34.6        hb03c661_0            conda-forge
  ca-certificates          2026.1.4      hbd8a1cb_0            conda-forge
  certifi                  2026.1.4      pyhd8ed1ab_0          conda-forge
  cffi                     2.0.0         py313hf46b229_1       conda-forge
  charset-normalizer       3.4.4         pyhd8ed1ab_0          conda-forge
  contourpy                1.3.3         py313hc8edb43_4       conda-forge
  curl                     8.18.0        h4e3cde8_0            conda-forge
  cycler                   0.12.1        pyhcf101f3_2          conda-forge
  diamond                  2.1.22        h13889ed_0            bioconda   
  entrez-direct            24.0          he881be0_0            bioconda   
  fonttools                4.61.1        py313h3dea7bd_0       conda-forge
  libuuid                  2.41.3        h5347b49_0            conda-forge
  libwebp-base             1.6.0         hd42ef1d_0            conda-forge
  libxcb                   1.17.0        h8a09558_0            conda-forge
  libxcrypt                4.4.36        hd590300_1            conda-forge
  libzlib                  1.3.1         hb9d3cd8_2            conda-forge
  matplotlib-base          3.10.8        py313h683a580_0       conda-forge
  mpi                      1.0           openmpi               conda-forge
  munkres                  1.1.4         pyhd8ed1ab_1          conda-forge
  ncbi-amrfinderplus       4.2.7         hf69ffd2_0            bioconda   
  ncbi-vdb                 3.3.0         h9948957_0            bioconda   
  ncurses                  6.5           h2d0b736_3            conda-forge
  numpy                    2.4.2         py313hf6604e3_1       conda-forge
  openjpeg                 2.5.4         h55fea9a_0            conda-forge
  openmpi                  4.1.6         hc5af2df_101          conda-forge
  openssl                  3.6.1         h35e630c_1            conda-forge
  packaging                26.0          pyhcf101f3_0          conda-forge
  pandas                   3.0.0         py313hbfd7664_0       conda-forge
  pbzip2                   1.1.13        h1fcc475_2            conda-forge
  pcre2                    10.45         hc749103_0            conda-forge
  perl                     5.32.1        7_hd590300_perl5      conda-forge
  perl-archive-tar         3.04          pl5321hdfd78af_0      bioconda   
  perl-carp                1.50          pl5321hd8ed1ab_0      conda-forge
  perl-common-sense        3.75          pl5321hd8ed1ab_0      conda-forge
  perl-compress-raw-bzip2  2.214         pl5321hda65f42_0      conda-forge
  perl-compress-raw-zlib   2.214         pl5321h4dac143_0      conda-forge
  perl-encode              3.21          pl5321hb9d3cd8_1      conda-forge
  perl-exporter            5.74          pl5321hd8ed1ab_0      conda-forge
  perl-exporter-tiny       1.002002      pl5321hd8ed1ab_0      conda-forge
  perl-extutils-makemaker  7.70          pl5321hd8ed1ab_0      conda-forge
  perl-io-compress         2.216         pl5321h503566f_0      bioconda   
  perl-io-zlib             1.15          pl5321hdfd78af_1      bioconda   
  perl-json                4.10          pl5321hdfd78af_1      bioconda   
  perl-json-xs             4.04          pl5321h9948957_0      bioconda   
  perl-list-moreutils      0.430         pl5321hdfd78af_0      bioconda   
  perl-list-moreutils-xs   0.430         pl5321h7b50bb2_5      bioconda   
  perl-parent              0.243         pl5321hd8ed1ab_0      conda-forge
  perl-pathtools           3.75          pl5321hb9d3cd8_2      conda-forge
  perl-scalar-list-utils   1.70          pl5321hb03c661_0      conda-forge
  perl-storable            3.15          pl5321hb9d3cd8_2      conda-forge
  perl-types-serialiser    1.01          pl5321hdfd78af_0      bioconda   
  pigz                     2.8           h421ea60_2            conda-forge
  piler-cr                 1.06          h9948957_6            bioconda   
  pillow                   12.1.0        py313h80991f8_0       conda-forge
  pip                      26.0.1        pyh145f28c_0          conda-forge
  psutil                   7.2.2         py313h54dd161_0       conda-forge
  pthread-stubs            0.4           hb9d3cd8_1002         conda-forge
  pycirclize               1.10.1        pyhd8ed1ab_0          conda-forge
  pycparser                2.22          pyh29332c3_1          conda-forge
  pyhmmer                  0.12.0        py313h366bbf7_0       bioconda   
  pyparsing                3.3.2         pyhcf101f3_0          conda-forge
  pyrodigal                3.7.0         py313h366bbf7_0       bioconda   
  pysocks                  1.7.1         pyha55dd90_7          conda-forge
  python                   3.13.12       hc97d973_100_cp313    conda-forge
  python-dateutil          2.9.0.post0   pyhe01879c_2          conda-forge
  python-isal              1.8.0         py313h07c4f96_1       conda-forge
  python-zlib-ng           1.0.0         py313ha345abf_1       conda-forge
  python_abi               3.13          8_cp313               conda-forge
  pyyaml                   6.0.3         py313h3dea7bd_1       conda-forge
  qhull                    2020.2        h434a139_5            conda-forge
  readline                 8.3           h853b02a_0            conda-forge
  requests                 2.32.5        pyhcf101f3_1          conda-forge
  six                      1.17.0        pyhe01879c_1          conda-forge
  tk                       8.6.13        noxft_h366c992_103    conda-forge
  trnascan-se              2.0.12        pl5321h7b50bb2_2      bioconda   
  tzdata                   2025c         hc9c84f9_1            conda-forge
  urllib3                  2.6.3         pyhd8ed1ab_0          conda-forge
  wget                     1.25.0        hc2d8bac_0            conda-forge
  xopen                    2.0.2         pyh707e725_2          conda-forge
  xorg-libxau              1.0.12        hb03c661_1            conda-forge
  xorg-libxdmcp            1.1.5         hb03c661_1            conda-forge
  yaml                     0.2.5         h280c20c_3            conda-forge
  zlib                     1.3.1         hb9d3cd8_2            conda-forge
  zlib-ng                  2.3.3         hceb46e0_1            conda-forge
  zstandard                0.25.0        py313h54dd161_1       conda-forge
  zstd                     1.5.7         hb78ec9c_6            conda-forge
```
</details>
</br>

Code repository: https://github.com/oschwengers/bakta

Basic information on how to use this tool:
- executable: bakta
- help: -h
- version: -v
- description: Annotates bacterial genomes

> Bakta is a tool for the rapid & standardized annotation of bacterial genomes and plasmids from both isolates and MAGs. It provides dbxref-rich, sORF-including and taxon-independent annotations in machine-readable JSON & bioinformatics standard file formats for automated downstream analysis.

Additional information:

WARNING : Image does not contain a database. 
  
Full documentation: https://github.com/oschwengers/bakta/

## Example Usage

```bash
bakta sample.fasta --db <path to database> --threads 20 --prefix sample
```
